index.js

```diff
@@ -711,11 +711,12 @@ var torrentStream = function (link, opts, cb) {
   }
 
   var removeTorrent = function (cb) {
+    engine.destroy()
     fs.unlink(torrentPath, function (err) {
-      if (err) return cb(err)
+      if (err && typeof cb === 'function') return cb(err)
```

Comment:
If `cb` is undefined, is it worth continuing? Maybe just initialize `cb` to a default value? E.g. `cb = cb || noop;`